smartfs.lua

```diff
@@ -862,26 +929,20 @@ smartfs.element("checkbox", {
 		self.data.label = self.data.label or ""
 	end,
 	build = function(self)
-		if self.data.value then
-			self.data.value = "true"
-		else
-			self.data.value = "false"
-		end
 		return "checkbox["..
 			self.data.pos.x..","..self.data.pos.y..
 			";"..
-			self.name..
+			self:getAbsName()..
 			";"..
 			minetest.formspec_escape(self.data.label)..
 			";" .. boolToStr(self.data.value) .."]"
 	end,
-	submit = function(self, fields, player)
-		if fields[self.name] then
-			-- self.data.value already set by value transfer
-			-- call the toggle function if defined
-			if self._tog then
-				self:_tog(self.root, player)
-			end
+	submit = function(self, field, player)
+		-- self.data.value already set by value transfer, but as string
+		self.data.value = minetest.is_yes(field)
```

Comment:
I see the line is not complied with principle "apply values before actions".
Hm, I check the implementation against this. Probably the _sfs_process_value_ and _sfs_process_action_ needs to be moved to element methods instead of state methods to be redefinable. This shoud take the recursion to be simpler implemented.